xdsl/dialects/transform.py

```diff
@@ -164,11 +164,15 @@ class ApplyRegisteredPassOp(IRDLOperation):
 
     name = "transform.apply_registered_pass"
 
-    options = prop_def(StringAttr, default_value=StringAttr(""))
+    options = opt_attr_def(StringAttr, default_value=StringAttr(""))
     pass_name = prop_def(StringAttr)
     target = operand_def(TransformHandleType)
     result = result_def(TransformHandleType)
 
+    assembly_format = (
+        "$pass_name `to` $target attr-dict `:` functional-type(operands, results)"
+    )
+
```

Comment:
Does this work?
```suggestion
    options = prop_def(StringAttr, default_value=StringAttr(""))
    pass_name = prop_def(StringAttr)
    target = operand_def(TransformHandleType)
    result = result_def(TransformHandleType)

    assembly_format = (
        "$pass_name `to` $target attr-dict `:` functional-type(operands, results)"
    )

    irdl_options = [ParsePropInAttrDict()]
```

Reply:
No:

> xdsl.utils.exceptions.PyRDLOpDefinitionError: ('Error during the parsing of the assembly format: ', (Span[72:72](text=''), "properties cannot be specified in the declarative format when 'ParsePropInAttrDict' IRDL option is used. They are instead parsed from the attribute dictionary."))

Reply:
CC @alexarice what do you think is the move here? It feels like we should just print the remaining props in the attr-dict, and not have it be all or nothing.

Reply:
This is quite annoying, I agree after reading the docs that it likely shouldn't be all or nothing, but this will require a refactor of the assembly format directives to allow some properties to appear in the attr-dict (and likely make sure that these properties are optional/default valued).

I think it's likely not correct to use an attribute instead of property here, and probably is also not correct to use `ParsePropInAttrDict`, but you would need to check the generic format of the mlir op to be sure.

Reply:
Can confirm that mlir prints `options` as a property:
```mlir
"builtin.module"() ({
  %0 = "test.op"() : () -> !transform.op<"builtin.module">
  %1 = "transform.apply_registered_pass"(%0) <{options = "foo", pass_name = "foo"}> : (!transform.op<"builtin.module">) -> !transform.op<"builtin.module">
}) : () -> ()
```

Reply:
@superlopuh just rebased this. Thanks for the review!